src/modalities/models/gpt2/gpt2_model.py

```diff
@@ -831,22 +854,28 @@ def __init__(
                             attention_config=attention_config,
                             dropout=dropout,
                             ffn_hidden=ffn_hidden,
-                            attention_norm=deepcopy(attention_norm),
-                            ffn_norm=deepcopy(ffn_norm),
+                            # deepcopy did not work here! The weights were then automatically
+                            # moved to a cuda device even when the deepcopied weights were on
+                            # a meta device!
+                            attention_norm=attention_norm_config.norm_type.value(**dict(attention_norm_config.config)),
+                            ffn_norm=ffn_norm_config.norm_type.value(**dict(ffn_norm_config.config)),
                         )
                         for _ in range(n_layer)
                     ]
                 ),
-                lm_head_norm=lm_head_norm,
+                lm_head_norm=lm_head_norm_config.norm_type.value(**dict(lm_head_norm_config.config)),
+                lm_head=nn.Linear(in_features=n_embd, out_features=vocab_size, bias=False),
```

Comment:
Should bias not be configurable?

Reply:
I would leave it as is for now. Before, we were tying lm_head with the embedding weights. In this case, there was also no bias term.